### ISSUE TITLE
Configuration for rule "no-use-before-define" is invalid: Value "true" is the wrong type

### ISSUE THREAD

**Issue by pdehaan** [mozilla/fxa-content-server#2599]
### Steps to reproduce:
1. Update to the latest `eslint` (I'm using the newly released [v0.23.0](http://eslint.org/blog/2015/06/eslint-0.23.0-released/) [globally installed]).
2. `$ eslint .`

### Actual results:
```
Error: /Users/pdehaan/dev/fxa-content-server_pd/fxa-content-server/.eslintrc:
	Configuration for rule "no-use-before-define" is invalid:
	Value "true" is the wrong type.
```
([source](https://github.com/mozilla/fxa-content-server/blob/c089ae6bd770f1219c7d3650d675d740e354dae4/.eslintrc#L38))

```js
    "no-use-before-define": true,
```

### Expected results:
Not sure if `true` should be a `1` (warning), or `2` (error). But a Boolean looks wrong and eslint@0.23.0 isn't happy.

**Comment by pdehaan** [comment]
**PROTIP:** Lets update the **grunt-eslint** devDep since a new version was released very recently to align w/ **eslint@0.23.0**.

The old **grunt-eslint** silently ate these errors, whereas the new **grunt-eslint** will implode with the power of 1000 suns when we have invalid rules/configs:

```
$ npm ls | grep eslint
├─┬ grunt-eslint@15.0.0
│ └─┬ eslint@0.23.0

$ grunt eslint
Running "eslint:grunt" (eslint) task
Warning: /Users/pdehaan/dev/fxa-content-server_pd/fxa-content-server/.eslintrc:
	Configuration for rule "no-use-before-define" is invalid:
	Value "true" is the wrong type.
 Use --force to continue.

Aborted due to warnings.
```

**Comment by TDA** [comment]
@pdehaan whoa, 1000 suns? Sure looks like it :O I'll start fixing this up :)
